### PR TITLE
Add JMeter example application

### PR DESCRIPTION
**Pull request by davidkopp**
This example demonstrates how to use JMeter to execute a test plan that makes HTTP requests to a web application.
[Apache JMeter](https://jmeter.apache.org/) is a load and performance testing tool.

JMeter Docker image: https://hub.docker.com/r/t2project/jmeter
Example web application: [T2-Project](https://t2-documentation.readthedocs.io)

### CODE REVIEW

File: jmeter/compose.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      POSTGRES_USER: postgres
+      POSTGRES_PASSWORD: postgres
+    networks:
+      - t2-network
+  gcb-mongo:
+    image: mongo:5.0
+    networks:
+      - t2-network
+  gcb-backend:
+    image: t2project/modulith:gmt-jmeter-example
```
Comment (ArneTR) [inline jmeter/compose.yml:16]: can you pin this with a hash

File: jmeter/compose.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      - gcb-postgres
+      - gcb-mongo
+    environment:
+      MONGO_HOST: gcb-mongo
+      POSTGRES_HOST: gcb-postgres
+      SPRING_PROFILES_ACTIVE: gmt
+    networks:
+      - t2-network
+  gcb-jmeter:
+    image: t2project/jmeter:5.6.2
```
Comment (ArneTR) [inline jmeter/compose.yml:27]: can you pin this with a hash

File: jmeter/jmeter-test-plan.jmx [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+          <stringProp name="IfController.condition">${logging_enabled}</stringProp>
+        </IfController>
+        <hashTree>
+          <JSR223Sampler guiclass="TestBeanGUI" testclass="JSR223Sampler" testname="LOG GMT_SCI_R" enabled="true">
+            <stringProp name="cacheKey">true</stringProp>
+            <stringProp name="filename"></stringProp>
+            <stringProp name="parameters"></stringProp>
+            <stringProp name="script">def currentTimeMicroseconds = System.currentTimeMillis() * 1000
+
+log.info(&quot;${currentTimeMicroseconds} GMT_SCI_R=1&quot;)</stringProp>
```
Comment (ArneTR) [inline jmeter/jmeter-test-plan.jmx:318]: Love it!

**Review by ArneTR — CHANGES_REQUESTED** [review]
Hey @davidkopp 

The flow of great PRs keeps coming. I love it!

Again a very high quality PR!!

When we take in examples from externals we always like to pin the depencies with a hash. This keeps the security issues lower and also increases reproducibility for people using your example.
I made two requests at the corresponding locations.

Hope that is ok to request!

If you however had in mind to use the example to track the changes over time by only referencing the github here, we can also talk about bringing that into our timeline project. Let me know! - https://metrics.green-coding.berlin/energy-timeline.html

**Comment by davidkopp** [comment]
It's ok for me to pin the images with hashes :)

**Comment by ArneTR** [comment]
lgtm

**Comment by ArneTR** [comment]
@dan-mm Can you have a look at this if it gets picked up by our example applications routine testing. Having a java jmeter application in there is very valueable.

Also please report how long it takes for this test to run on our infrastructure

**Comment by dan-mm** [comment]
@ArneTR - tested it out. It does indeed get picked up by the example applications, I just ran the tests manually and saw it picked up and pass.

To answer the time question, its about 1 - 2 minutes. Considering the entire example apps test suite takes 3.5 hours its nothing. I double checked by pulling and running the smoketest on this directory alone, it takes between 53 seconds - 2 minutes to complete.

**Pull request merged**